include/drivers/gpio.h

```diff
@@ -25,6 +25,12 @@ struct gpio_driver {
 	const struct gpio_driver_ops ops;
 };
 
+struct gpio_pin {
```

Comment:
Please make this:
```c
struct gpio_handle {
	struct device *const dev;
	const uint8_t pin;
	const uint8_t mode;
};
```

This will allow a bit better optimization by GCC and will match the `struct clock_handle` in #120.

Reply:
Done.